api/server/server.go

```diff
@@ -1642,6 +1643,17 @@ func ServeApi(job *engine.Job) engine.Status {
 	return engine.StatusOK
 }
 
+func removeUnixSocket(sock string) {
+	parts := strings.SplitN(sock, "://", 2)
+	if parts[0] == "unix" {
+		err := os.Remove(parts[1])
+		if err != nil {
```

Comment:
if it doesn't exist already (i.e. failed to start server etc and there's nothing at this path) you'll get an err here (os.IsNotExist) and you'll be reporting that as an error, which is not really a failure.